libraries/botbuilder-dialogs-adaptive/src/input/confirmInput.ts

```diff
@@ -97,11 +96,7 @@ export class ConfirmInput extends InputDialog {
 
     protected async onRenderPrompt(dc: DialogContext, state: InputState): Promise<Partial<Activity>> {
         // Determine locale
-        let locale: string = dc.context.activity.locale || this.defaultLocale.getValue(dc.state);
-        if (!locale || !ConfirmInput.defaultChoiceOptions.hasOwnProperty(locale)) {
-            locale = 'en-us';
-        }
-        locale = locale.toLowerCase(); // to match format 'en-US'
+        let locale: string = this.determineCulture(dc);
```

Comment:
There are a handful of places in this PR where a `string` type is manually assigned even though the type can be inferred. We should avoid this - let the compiler do as much heavy lifting as it can.

```typescript
let locale = this.determineCulture(dc);
```